lib/puppet/type/puppet_authorization_hocon_rule.rb

```diff
@@ -68,20 +61,14 @@ def change_to_s(current, new)
       real_new << new
       real_new.flatten!
       real_new.uniq!
-      "value changed [#{Array(current).flatten.join(", ")}] to [#{real_new.join(", ")}]"
+      "value changed [#{Array(current).flatten.join(', ')}] to [#{real_new.join(', ')}]"
     end
   end
 
   validate do
-    message = ""
-    if self.original_parameters[:path].nil?
-      message += "path is a required parameter. "
-    end
-    if self.original_parameters[:value].nil? && self[:ensure] != :absent
-      message += "value is a required parameter unless ensuring a setting is absent."
-    end
-    if message != ""
-      raise(Puppet::Error, message)
-    end
+    message = ''
+    message += 'path is a required parameter. ' if original_parameters[:path].nil?
+    message += 'value is a required parameter unless ensuring a setting is absent.' if original_parameters[:value].nil? && self[:ensure] != :absent
```

Comment:
Feels like this should be
```suggestion
    if original_parameters[:path].nil?
      message += 'path is a required parameter.'
      message += ' value is a required parameter unless ensuring a setting is absent.' if self[:ensure] != :absent
    end
```

Reply:
That behaves differently.  In the original, the first message gates on `:path` and the second on gates on `:value`.  In the suggested change, they both gate on `:path` and `:value` isn't considered.